kafka/admin/client.py

```diff
@@ -1000,22 +1002,47 @@ def _describe_consumer_groups_process_response(self, response):
         """Process a DescribeGroupsResponse into a group description."""
         if response.API_VERSION <= 3:
             assert len(response.groups) == 1
-            # TODO need to implement converting the response tuple into
-            # a more accessible interface like a namedtuple and then stop
-            # hardcoding tuple indices here. Several Java examples,
-            # including KafkaAdminClient.java
-            group_description = response.groups[0]
-            error_code = group_description[0]
+            for response_field, response_name in zip(response.SCHEMA.fields, response.SCHEMA.names):
+                if type(response_field) == Array:
+                    described_groups = response.__dict__[response_name]
+                    described_groups_field_schema = response_field.array_of
+                    for described_group  in described_groups:
+                        described_group_information_list = []
+                        is_consumer_protocol_type = False
+                        for (described_group_information, group_information_name, group_information_field) in zip(described_group, described_groups_field_schema.names, described_groups_field_schema.fields):
+                            if group_information_name == 'protocol_type':
+                                protocol_type = described_group_information
+                                is_consumer_protocol_type = (protocol_type == ConsumerProtocol.PROTOCOL_TYPE or not protocol_type)
```

Comment:
Thanks for pointing this out. I reversed the order to the below in the next change:
"not protocol_type or protocol_type == ConsumerProtocol.PROTOCOL_TYPE"
Basically its checking if the protocol_type returned by the broker is an empty string or if it is "consumer" then it needs to execute the extra decoding of the member metadata and member assignment fields in the member array.
The Kafka Java implementation for this:
https://github.com/apache/kafka/blob/d63eaaaa0181bb7b9b4f5ed088abc00d7b32aeb0/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L2881
Hope it helps. Please let me know if you need more information.

Reply:
makes sense. One suggestion--for your github links, hit the `y` key on your keyboard and it will tie the link to a specific commit, rather than worrying about `trunk` changing under your feet.

Reply:
Thanks for the tip! I will use that next time. 😃